tiledb/sm/query/dense_reader.cc

```diff
@@ -656,7 +656,8 @@ Status DenseReader::read_attributes(
       dst_off_bufs.push_back((uint8_t*)buffers_[name].buffer_);
       dst_var_bufs.push_back((uint8_t*)buffers_[name].buffer_var_);
       dst_val_bufs.push_back(buffers_[name].validity_vector_.buffer());
-      attributes.push_back(array_schema_->attribute(name));
+      attributes.push_back(
+          tdb::make_shared<Attribute>(HERE(), array_schema_->attribute(name)));
```

Comment:
What's the reason for a new allocation here? The lifespan of the reader won't exceed that of the array schema, unless I'm missing something.